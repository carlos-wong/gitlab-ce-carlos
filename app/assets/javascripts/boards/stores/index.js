import Vue from 'vue';
import Vuex from 'vuex';
import state from 'ee_else_ce/boards/stores/state';
import getters from 'ee_else_ce/boards/stores/getters';
import actions from 'ee_else_ce/boards/stores/actions';
import mutations from 'ee_else_ce/boards/stores/mutations';

Vue.use(Vuex);

export const createStore = () =>
  new Vuex.Store({
    state,
    getters,
    actions,
    mutations,
  });

export default createStore();
