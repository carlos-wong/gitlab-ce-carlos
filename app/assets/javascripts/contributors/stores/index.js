import Vue from 'vue';
import Vuex from 'vuex';
import state from './state';
import mutations from './mutations';
import * as getters from './getters';
import * as actions from './actions';

Vue.use(Vuex);

export const createStore = () =>
  new Vuex.Store({
    actions,
    mutations,
    getters,
    state: state(),
  });

export default createStore();
