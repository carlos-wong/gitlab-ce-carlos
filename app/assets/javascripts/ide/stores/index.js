import Vue from 'vue';
import Vuex from 'vuex';
import state from './state';
import * as actions from './actions';
import * as getters from './getters';
import mutations from './mutations';
import commitModule from './modules/commit';
import pipelines from './modules/pipelines';
import mergeRequests from './modules/merge_requests';
import branches from './modules/branches';
import fileTemplates from './modules/file_templates';
import paneModule from './modules/pane';
import clientsideModule from './modules/clientside';

Vue.use(Vuex);

export const createStore = () =>
  new Vuex.Store({
    state: state(),
    actions,
    mutations,
    getters,
    modules: {
      commit: commitModule,
      pipelines,
      mergeRequests,
      branches,
      fileTemplates: fileTemplates(),
      rightPane: paneModule(),
      clientside: clientsideModule(),
    },
  });

export default createStore();
