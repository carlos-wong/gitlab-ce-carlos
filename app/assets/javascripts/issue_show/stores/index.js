import _ from 'underscore';
import { convertObjectPropsToCamelCase } from '~/lib/utils/common_utils';
import updateDescription from '../utils/update_description';

export default class Store {
  constructor(initialState) {
    this.state = initialState;
    this.formState = {
      title: '',
      description: '',
      lockedWarningVisible: false,
      updateLoading: false,
      lock_version: 0,
      issuableTemplates: [],
    };
  }

  updateState(data) {
    if (this.stateShouldUpdate(data)) {
      this.formState.lockedWarningVisible = true;
    }

    Object.assign(this.state, convertObjectPropsToCamelCase(data));
    // find if there is an open details node inside of the issue description.
    const descriptionSection = document.body.querySelector(
      '.detail-page-description.content-block',
    );
    const details =
      !_.isNull(descriptionSection) && descriptionSection.getElementsByTagName('details');

    this.state.descriptionHtml = updateDescription(data.description, details);
    this.state.titleHtml = data.title;
    this.state.lock_version = data.lock_version;
  }

  stateShouldUpdate(data) {
    return (
      this.state.titleText !== data.title_text ||
      this.state.descriptionText !== data.description_text
    );
  }

  setFormState(state) {
    this.formState = Object.assign(this.formState, state);
  }
}
