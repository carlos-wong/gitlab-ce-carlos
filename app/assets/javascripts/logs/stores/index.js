import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import * as getters from './getters';
import mutations from './mutations';
import state from './state';

Vue.use(Vuex);

export const createStore = () =>
  new Vuex.Store({
    modules: {
      environmentLogs: {
        namespaced: true,
        actions,
        mutations,
        state: state(),
        getters,
      },
    },
  });

export default createStore;
