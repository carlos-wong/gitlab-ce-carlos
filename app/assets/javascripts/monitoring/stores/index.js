import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import * as getters from './getters';
import mutations from './mutations';
import state from './state';

Vue.use(Vuex);

export const createStore = () =>
  new Vuex.Store({
    modules: {
      monitoringDashboard: {
        namespaced: true,
        actions,
        getters,
        mutations,
        state,
      },
    },
  });

export default createStore();
