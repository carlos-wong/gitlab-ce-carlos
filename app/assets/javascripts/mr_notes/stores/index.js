import Vue from 'vue';
import Vuex from 'vuex';
import notesModule from '~/notes/stores/modules';
import diffsModule from '~/diffs/store/modules';
import mrPageModule from './modules';

Vue.use(Vuex);

export const createStore = () =>
  new Vuex.Store({
    modules: {
      page: mrPageModule(),
      notes: notesModule(),
      diffs: diffsModule(),
    },
  });

export default createStore();
