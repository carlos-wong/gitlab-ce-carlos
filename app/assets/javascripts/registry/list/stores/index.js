import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import * as getters from './getters';
import mutations from './mutations';
import createState from './state';

Vue.use(Vuex);

export default new Vuex.Store({
  state: createState(),
  actions,
  getters,
  mutations,
});
