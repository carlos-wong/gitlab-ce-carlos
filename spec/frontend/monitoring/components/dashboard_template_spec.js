import { shallowMount } from '@vue/test-utils';
import MockAdapter from 'axios-mock-adapter';
import axios from '~/lib/utils/axios_utils';
import Dashboard from '~/monitoring/components/dashboard.vue';
import { createStore } from '~/monitoring/stores';
import { propsData } from '../init_utils';

jest.mock('~/lib/utils/url_utility');

describe('Dashboard template', () => {
  let wrapper;
  let store;
  let mock;

  beforeEach(() => {
    store = createStore();
    mock = new MockAdapter(axios);
  });

  afterEach(() => {
    if (wrapper) {
      wrapper.destroy();
      wrapper = null;
    }
    mock.restore();
  });

  it('matches the default snapshot', () => {
    wrapper = shallowMount(Dashboard, {
      propsData: { ...propsData },
      methods: {
        fetchData: jest.fn(),
      },
      store,
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
