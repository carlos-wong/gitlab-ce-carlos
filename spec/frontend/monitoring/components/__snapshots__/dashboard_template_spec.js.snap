// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard template matches the default snapshot 1`] = `
<div
  class="prometheus-graphs"
  data-qa-selector="prometheus_graphs"
>
  <div
    class="prometheus-graphs-header gl-p-3 pb-0 border-bottom bg-gray-light"
  >
    <div
      class="row"
    >
      <gl-form-group-stub
        class="col-sm-12 col-md-6 col-lg-2"
        label="Dashboard"
        label-for="monitor-dashboards-dropdown"
        label-size="sm"
      >
        <dashboards-dropdown-stub
          class="mb-0 d-flex"
          defaultbranch="master"
          id="monitor-dashboards-dropdown"
          selecteddashboard="[object Object]"
          toggle-class="dropdown-menu-toggle"
        />
      </gl-form-group-stub>
       
      <gl-form-group-stub
        class="col-sm-6 col-md-6 col-lg-2"
        label="Environment"
        label-for="monitor-environments-dropdown"
        label-size="sm"
      >
        <gl-dropdown-stub
          class="mb-0 d-flex"
          data-qa-selector="environments_dropdown"
          id="monitor-environments-dropdown"
          menu-class="monitor-environment-dropdown-menu"
          text="production"
          toggle-class="dropdown-menu-toggle"
        >
          <div
            class="d-flex flex-column overflow-hidden"
          >
            <gl-dropdown-header-stub
              class="monitor-environment-dropdown-header text-center"
            >
              Environment
            </gl-dropdown-header-stub>
             
            <gl-dropdown-divider-stub />
             
            <gl-search-box-by-type-stub
              class="m-2"
              value=""
            />
             
            <div
              class="flex-fill overflow-auto"
            />
             
            <div
              class="text-secondary no-matches-message"
            >
              
              No matching results
            
            </div>
          </div>
        </gl-dropdown-stub>
      </gl-form-group-stub>
       
      <gl-form-group-stub
        class="col-sm-6 col-md-6 col-lg-4"
        label="Show last"
        label-for="monitor-time-window-dropdown"
        label-size="sm"
      >
        <date-time-picker-stub
          options="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
          value="[object Object]"
        />
      </gl-form-group-stub>
       
      <!---->
    </div>
  </div>
   
  <empty-state-stub
    clusterspath="/path/to/clusters"
    documentationpath="/path/to/docs"
    emptygettingstartedsvgpath="/path/to/getting-started.svg"
    emptyloadingsvgpath="/path/to/loading.svg"
    emptynodatasmallsvgpath="/path/to/no-data-small.svg"
    emptynodatasvgpath="/path/to/no-data.svg"
    emptyunabletoconnectsvgpath="/path/to/unable-to-connect.svg"
    selectedstate="gettingStarted"
    settingspath="/path/to/settings"
  />
</div>
`;
